build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.code-quality.gradle.kts

```diff
@@ -91,17 +100,21 @@ tasks.withType<CodeNarc>().configureEach {
 val SourceSet.allGroovy: SourceDirectorySet
     get() = withConvention(GroovySourceSet::class) { allGroovy }
 
-abstract class CodeNarcRule : ComponentMetadataRule {
+abstract class CodeNarcRule @Inject constructor(
+    private val groovyVersion: String
+): ComponentMetadataRule {
     override fun execute(context: ComponentMetadataContext) {
         context.details.allVariants {
             withDependencies {
-                removeAll { it.group == "org.codehaus.groovy" }
-                add("org.codehaus.groovy:groovy") {
-                    version { prefer(groovy.lang.GroovySystem.getVersion()) }
+                val isAtLeastGroovy4 = VersionNumber.parse(groovyVersion).major >= 4
+                val groovyGroup = if(isAtLeastGroovy4) "org.apache.groovy" else "org.codehaus.groovy"
+                removeAll { listOf("org.apache.groovy", "org.codehaus.groovy").contains(it.group) }
```

Comment:
Why is this not using groovyGroup?

Reply:
I'll change it to read: `removeAll { it.group == groovyGroup }`